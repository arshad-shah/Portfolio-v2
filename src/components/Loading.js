import React from "react";
import { styled } from "@mui/material/styles";
import { CircularProgress, Typography } from "@mui/material";
const PREFIX = "Loading";

const classes = {
  root: `${PREFIX}-root`,
  name: `${PREFIX}-name`,
  progress: `${PREFIX}-progress`,
};

const Root = styled("div")(({ theme }) => ({
  [`&.${classes.root}`]: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    alignContent: "center",
    flexWrap: "nowrap",
    backgroundColor: "#ffffff",
    minHeight: "100vh",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
    backgroundPosition: "fixed",
  },

  [`& .${classes.name}`]: {
    padding: "1rem",
  },

  [`& .${classes.progress}`]: {
    padding: "1rem",
  },
}));

export default function Loading() {
  return (
    <Root className={classes.root}>
      <CircularProgress
        className={classes.progress}
        color="secondary"
        size={100}
        thickness={5}
      />
      <Typography className={classes.name} variant="h3">
        Loading....
      </Typography>
    </Root>
  );
}
