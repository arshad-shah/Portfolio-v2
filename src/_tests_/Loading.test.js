import React from "react";
import { render } from "@testing-library/react";
import Loading from "../components/Loading";

describe("Header of the site:", () => {
  it("should render Header", () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
