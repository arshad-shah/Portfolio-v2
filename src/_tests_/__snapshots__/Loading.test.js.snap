// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header of the site: should render Header 1`] = `
<div>
  <div
    class="Loading-root css-1qz95ku"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary Loading-progress css-1szo49z-MuiCircularProgress-root"
      role="progressbar"
      style="width: 100px; height: 100px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="19.5"
          stroke-width="5"
        />
      </svg>
    </span>
    <h3
      class="MuiTypography-root MuiTypography-h3 Loading-name css-gepadz-MuiTypography-root"
    >
      Loading....
    </h3>
  </div>
</div>
`;
